pkg/azuredx/client/client.go

```diff
@@ -18,27 +20,49 @@ import (
 
 type AdxClient interface {
 	TestRequest(ctx context.Context, datasourceSettings *models.DatasourceSettings, properties *models.Properties, additionalHeaders map[string]string) error
-	KustoRequest(ctx context.Context, url string, payload models.RequestPayload, additionalHeaders map[string]string) (*models.TableResponse, error)
+	KustoRequest(ctx context.Context, cluster string, url string, payload models.RequestPayload, additionalHeaders map[string]string) (*models.TableResponse, error)
+	ARGClusterRequest(ctx context.Context, payload models.ARGRequestPayload, additionalHeaders map[string]string) ([]models.ClusterOption, error)
 }
 
 var _ AdxClient = new(Client) // validates interface conformance
 
 // Client is an http.Client used for API requests.
 type Client struct {
-	httpClient *http.Client
+	httpClientKusto      *http.Client
+	httpClientManagement *http.Client
 }
 
 // NewClient creates a Grafana Plugin SDK Go Http Client
 func New(ctx context.Context, instanceSettings *backend.DataSourceInstanceSettings, dsSettings *models.DatasourceSettings, azureSettings *azsettings.AzureSettings, credentials azcredentials.AzureCredentials) (*Client, error) {
-	httpClient, err := newHttpClient(ctx, instanceSettings, dsSettings, azureSettings, credentials)
+	httpClientAzureCloud, err := newHttpClientAzureCloud(ctx, instanceSettings, dsSettings, azureSettings, credentials)
 	if err != nil {
 		return nil, err
 	}
-	return &Client{httpClient: httpClient}, nil
+	httpClientManagement, err := newHttpClientManagement(ctx, instanceSettings, dsSettings, azureSettings, credentials)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{httpClientKusto: httpClientAzureCloud, httpClientManagement: httpClientManagement}, nil
 }
 
 // TestRequest handles a data source test request in Grafana's Datasource configuration UI.
 func (c *Client) TestRequest(ctx context.Context, datasourceSettings *models.DatasourceSettings, properties *models.Properties, additionalHeaders map[string]string) error {
+	if err := c.testManagementClientAndGetCluster(ctx, datasourceSettings, additionalHeaders); err != nil {
+		return err
+	}
+
+	if datasourceSettings.ClusterURL == "" {
+		return nil
+	}
+
+	if err := c.testKustoClientWithDefaultCluster(ctx, datasourceSettings, datasourceSettings.ClusterURL, properties, additionalHeaders); err != nil {
```

Comment:
One other final nit 😜 what happens here if there is no `ClusterURL` now that it's an optional property